src/libmongoc/doc/mongoc_database_find_collections.rst

```diff
@@ -43,5 +43,9 @@ Errors are propagated via the ``error`` parameter.
 Returns
 -------
 
-A cursor where each result corresponds to the server's representation of a collection in this database.
+.. include:: includes/returns-cursor.txt
```

Comment:
This may be a case to not use the include directly.

Part of the deprecated behavior of this function is that it returns `NULL` if an error occurred:
https://github.com/mongodb/mongo-c-driver/blob/6bacf922669ab780fa5fb37e9c561b8709e27e29/src/libmongoc/src/mongoc/mongoc-database.c#L758:L758

Instead of using the `returns-cursor.txt` directly, this might be a special case, and to paste everything except for the `The returned mongoc_cursor_t is never NULL, even on error.` sentence:


```suggestion
This function returns a newly allocated :symbol:`mongoc_cursor_t` that should be freed with :symbol:`mongoc_cursor_destroy()` when no longer in use. The user must call :symbol:`mongoc_cursor_next()` on the returned :symbol:`mongoc_cursor_t` to execute the initial command.
```

Reply:
Maybe documentation should make explicit that in this case `NULL` can be returned in case of error? I mean something like this:

```suggestion
This function returns a newly allocated :symbol:`mongoc_cursor_t` that should be freed with :symbol:`mongoc_cursor_destroy()` when no longer in use, or `NULL` in case of error. The user must call :symbol:`mongoc_cursor_next()` on the returned :symbol:`mongoc_cursor_t` to execute the initial command.
```

What do you think?